kubectl-fdb/cmd/k8s_client.go

```diff
@@ -181,24 +181,34 @@ func getAllPodsFromClusterWithCondition(kubeClient client.Client, clusterName st
 	podNames := make([]string, 0, len(processesSet))
 	pods, err := getPodsForCluster(kubeClient, cluster)
 	if err != nil {
-		return podNames, err
+		return podNames, nil, err
 	}
 
+	podMap := make(map[string]corev1.Pod)
 	for _, pod := range pods.Items {
-		for process := range processesSet {
-			if pod.Status.Phase != corev1.PodRunning {
-				continue
-			}
-
-			if pod.Labels[cluster.GetProcessGroupIDLabel()] != string(process) {
-				continue
-			}
+		podMap[pod.Labels[cluster.GetProcessGroupIDLabel()]] = pod
+	}
 
-			podNames = append(podNames, pod.Name)
+	var warnings []string
+	for process := range processesSet {
+		if _, ok := podMap[string(process)]; !ok {
+			warnings = append(warnings, fmt.Sprintf("Process Group: %s, is missing pods.", process))
```

Comment:
```suggestion
			fmt.Fprintln(stdErr, ("Process Group: %s, is missing pods.", process)
```

With the changes from above.